### ISSUE TITLE
Regression bug: assets do not work with platform-specific extensions

### ISSUE THREAD

**Issue by kasl** [facebook/metro#676]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
I have platform-dependent pictures (with `.android.png | .ios.png` extension) in my project (metro version was 0.58) that successfully resolves. I decided to upgrade to the latest version of RN and got an error while building the bundle. I tried to repeat this on a clean project and it reproduces.

```
error: Error: Unable to resolve module ./images/pencil.png from /Users/Kasl/Documents/apps/RNAssetsResolveIssue/App.js: 

None of these files exist:
  * pencil.png
  * images/pencil.png/index(.native|.android.js|.native.js|.js|.android.json|.native.json|.json|.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.tsx)
  68 |         style={backgroundStyle}>
  69 |         <Header />
> 70 |         <Image source={require('./images/pencil.png')} />
     |                                 ^
  71 |         <View
  72 |           style={{
  73 |             backgroundColor: isDarkMode ? Colors.black : Colors.white,
    at ModuleResolver.resolveDependency (/Users/Kasl/Documents/apps/RNAssetsResolveIssue/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:211:15)
    at DependencyGraph.resolveDependency (/Users/Kasl/Documents/apps/RNAssetsResolveIssue/node_modules/metro/src/node-haste/DependencyGraph.js:413:43)
    at Object.resolve (/Users/Kasl/Documents/apps/RNAssetsResolveIssue/node_modules/metro/src/lib/transformHelpers.js:317:42)
    at resolve (/Users/Kasl/Documents/apps/RNAssetsResolveIssue/node_modules/metro/src/DeltaBundler/traverseDependencies.js:629:33)
    at /Users/Kasl/Documents/apps/RNAssetsResolveIssue/node_modules/metro/src/DeltaBundler/traverseDependencies.js:645:26
    at Array.reduce (<anonymous>)
    at resolveDependencies (/Users/Kasl/Documents/apps/RNAssetsResolveIssue/node_modules/metro/src/DeltaBundler/traverseDependencies.js:644:33)
    at /Users/Kasl/Documents/apps/RNAssetsResolveIssue/node_modules/metro/src/DeltaBundler/traverseDependencies.js:329:33
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/Users/Kasl/Documents/apps/RNAssetsResolveIssue/node_modules/metro/src/DeltaBundler/traverseDependencies.js:137:24)

```
<img width="306" alt="Снимок экрана 2021-06-10 в 18 53 44" src="https://user-images.githubusercontent.com/3302886/121557163-b6e24f80-ca3e-11eb-95ce-aa37cdf74555.png">

**If the current behavior is a bug, please provide the steps to reproduce and a minimal repository on GitHub that we can `yarn install` and `yarn test`.**
https://github.com/kasl/rn-assets-resolve-issue

**What is the expected behavior?**
I think metro should correctly resolve assets (pictures) with platform-specific extension.
https://reactnative.dev/docs/images
> The image name is resolved the same way JS modules are resolved. In the example above, the bundler will look for my-icon.png in the same folder as the component that requires it. Also, if you have my-icon.ios.png and my-icon.android.png, the bundler will pick the correct file for the platform.

**Please provide your exact Metro configuration and mention your Metro, node, yarn/npm version and operating system.**
node 12.22.1
yarn 1.22.10
metro 0.64.0
macOS 11.4

**Comment by motiz88** [comment]
This functionality was removed in 7674ca076e03483a17a37c31481807e2b67dea29 (released in [0.59.0](https://github.com/facebook/metro/releases/tag/v0.59.0) last year) so the docs are out of date. I've opened facebook/react-native-website#2653 to update the docs.

The recommended workaround is to use e.g. `Platform.select` to select the correct image for each platform.

**Comment by kasl** [comment]
It's sad. Thanks for the clear answer.
